core/modules/path/path.module

```diff
@@ -596,7 +596,7 @@ function path_get_info() {
   $all_info = backdrop_static(__FUNCTION__);
```

Comment:
This should be:
```
  $all_info = &backdrop_static(__FUNCTION__);
```

To use this design pattern, you need to use a reference assignment to `backdrop_static()` (see the `&`); otherwise, you don't get a proper reference to the static stuff inside.

There are several places in Backdrop core that the `&` is missing, but that will be a separate issue (to come).

Reply:
I believe that it is better to leave this PR as is, to address the identified code clean-up changes, and to implement the changes required for issue https://github.com/backdrop/backdrop-issues/issues/5142 in a new PR.

Reply:
Following further consideration, I have now suggested an improved code change that includes that "&" character as required.  Refer to https://github.com/backdrop/backdrop-issues/issues/5137#issuecomment-890411294